image-filters/image-filters.markdown

```diff
@@ -773,36 +771,37 @@ matrices of specific values that sharpen images.
 There are three main components to the app (\aosafigref{500l.imagefilters.architecture}).
 
 ### The App
-Consists of one file: `ImageFilterApp.java`. This extends `PApplet` (the
-Processing app superclass) and handles layout, user interaction etc. This class
+The app consists of one file: `ImageFilterApp.java`. This 
+extends `PApplet` (the
+Processing app superclass) and handles layout, user interaction, etc. This class
 is the hardest to test, so we want to keep it as small as possible.
 
 ### Model
-Consists of three files. `HSBColor.java` which is a simple container for HSB
-colors (consisting of hue, saturation, and brightness). `IFAImage` which is a
-wrapper around `PImage` for testability (`PImage` contains a number of final
-methods which cannot be mocked). Finally `ImageState.java` is the object
-describing the state of the image - what level of filters should be applied,
-and which, and handles loading the image (note: the image needs to be reloaded
+Model consists of three files: `HSBColor.java` is a simple container for HSB
+colors (consisting of hue, saturation, and brightness). `IFAImage` is a
+wrapper around `PImage` for testability. (`PImage` contains a number of final
+methods which cannot be mocked.) Finally, `ImageState.java` is the object
+which describes the state of the image — what level of filters should be applied,
+and which filters — and handles loading the image. (Note: The image needs to be reloaded
 whenever color filters are adjusted down, and whenever the dominant hue is
-recalculated. For clarity, we just reload each time the image is processed).
+recalculated. For clarity, we just reload each time the image is processed.)
 
 ### Color
-Consists of two files: `ColorHelper.java`, which is where all the image
-processing and filtering takes place. And `PixelColorHelper.java` which
+Color consists of two files: `ColorHelper.java` is where all the image
+processing and filtering takes place, and `PixelColorHelper.java` 
 abstracts out final `PApplet` methods for pixel colors for testability.
 
 \aosafigure[240pt]{image-filters-images/architecture.jpg}{Architecture diagram}{500l.imagefilters.architecture}
 
 ### Wrapper Classes and Tests
 Briefly mentioned above, there are two wrapper classes (`IFAImage` and
-`PixelColorHelper`) that wrap library methods for testability. This is because
+`PixelColorHelper`) that wrap library methods for testability. This is because,
 in Java, final methods are methods that cannot be overridden or hidden by
```

Comment:
I would rephrase this to say, "in Java, the keyword final indicates a method that cannot be overridden..."